internetarchive/openlibrary
# Separate copy-to-clipboard js + improve ux

Follow-up to #8806


Makes the following changes to the social share modal:
1. Closes the share modal when any share link is clicked.
2. Moves embed link's inline JS to Webpack.

Additionally, clicking the embed link will now copy the embed code to your clipboard and display a toast message.

### Technical

The "copy to clipboard on click" JS code has been generalized by these changes.  A new "copy to clipboard" affordance can be created by adding the following to any HTML element:

1. The `copy-to-clipboard` class
2. A `data-string-to-copy` attribute, which contains the string that will be copied to the clipboard when the element is clicked
3. A `data-i18n` attribute, which contains a JSON string with a `toast` property.  This is what will be displayed in the toast message after a string has been copied to the clipboard.

Usage example:
```html

<span class="copy-to-clipboard" data-string-to-copy="Hello world" data-i18n="$json_encode({'toast': 'Hello patron'})">Click to copy</span>
```

### Testing

1. Navigate to the book page and click the share icon.  Ensure that the share modal opens (this is BAU).
2. Click each share link and ensure that they continue to function as expected.  Ensure that the modal is also closed after each click.
3. Click the embed link.  Ensure that the embed code is copied to your clipboard.

### Screenshot


### Stakeholders
